sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -252,10 +254,14 @@ class HadoopTableReader(
         partProps.asScala.foreach {
           case (key, value) => props.setProperty(key, value)
         }
-        deserializer.initialize(hconf, props)
+        DeserializerLock.synchronized {
+          deserializer.initialize(hconf, props)
+        }
         // get the table deserializer
         val tableSerDe = localTableDesc.getDeserializerClass.getConstructor().newInstance()
-        tableSerDe.initialize(hconf, localTableDesc.getProperties)
```

Comment:
localTableDesc = tableDesc, and tableProperties = tableDesc.getProperties, which was used a little above in this same block, and might as well be used here.

Reply:
Actually, the `tableSerDe` can be instantiated and initialized before the transformation here. Then it will be done just once, instead of repeatedly for each Hive partition.
(The same applies to the `Deserializer` for the table in `makeRDDForTable`.)
I'll make a change.

Reply:
I was mistaken. The SerDe has to be instantiated and initialized within the transformation, as it is not Serializable and cannot be sent in the closure.